udf/extract_payload_uri.sql

```diff
@@ -0,0 +1,80 @@
+/*
+
+Extract all parts of the `uri` field used in payload tables. 
+
+*/
+CREATE TEMP FUNCTION udf_extract_payload_uri(uri STRING)
```

Comment:
It looks like we're moving towards a suite of uri parsing functions, so we should make sure they're named similarly for better discoverability. In particular, we have `parse_desktop_telemetry_uri` which may be a subset of the functionality here.

We need different functions for telemetry vs. structured; see https://github.com/mozilla/gcp-ingestion/blob/master/ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseUri.java#L56-L63

Reply:
The edge spec is also found in the data docs, for reference: https://docs.telemetry.mozilla.org/concepts/pipeline/http_edge_spec.html